### PR TITLE
Fix #3140  Shifting Instrumentation tests in Unit test directory for AdministratorControlsFragmentTest

### LINKED ISSUE #3140: Shift Instrumentation test written in Unit Test directory for AdministratorControlsFragmentTest
**Describe the bug**
There are some test cases written in `src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt` which are instrumentation test but the file exist in unit test folder. (i.e. app/src/test)

**Expected behavior**
Look for the instrumentation tests in this file and shift them to the instrumentation test directory (i.e. app/src/sharedTest). 

**Additional context**
[How to test on Espresso?](https://developer.android.com/codelabs/advanced-android-kotlin-training-testing-basics#4)

**How to send a Pull Request for this Issue**
Add screenshot showing shifted tests are passing on Espresso.

### PR DESCRIPTION
**Pull request by Rohit1173**
## Explanation
Fix #3140: Shifting tests which were instrumentation tests but written in the Unit test directory in the AdministratorControlsFragmentTest

Shifted tests Screenshot (Device(Emulator): Pixel-3A , API-28)
Screenshot 2022-01-25 125005

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by Rohit1173** [comment]
@yash10019coder  PTAL

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    Intents.release()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  @Test
+  fun testAdministratorControlsFragment_clickEditProfile_checkSendingTheCorrectIntent() {
+    ActivityScenario.launch<AdministratorControlsActivity>(
```
Comment (yash10019coder) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Add resource inputs for these static object references

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      createAdministratorControlsActivityIntent(
+        0
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      Espresso.onView(ViewMatchers.withId(R.id.edit_profiles_text_view))
```
Comment (yash10019coder) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Same goes here

**Review by yash10019coder — CHANGES_REQUESTED** [review]
Hi @Rohit1173 check for the static checks and also add a newline at the end of the file thanks

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt (hunk 1 above)
Comment (Rohit1173) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: done

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt (hunk 2 above)
Comment (Rohit1173) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: done

**Comment by yash10019coder** [comment]
@Rohit1173 please fix failing ci checks and also I think that this PR is blocked by #4029 what do you think about this @rt4914 PTAL thanks

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+  @Test
+  fun testAdministratorControlsFragment_clickEditProfile_checkSendingTheCorrectIntent() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        0
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(ViewMatchers.withId(R.id.edit_profiles_text_view))
```
Comment (yash10019coder) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Add resource imports for ViewMatchers and click performers in this and the following test

**Review by yash10019coder — CHANGES_REQUESTED** [review]
@Rohit1173 add resource imports and also fix failing test cases thanks

**Review by yash10019coder — APPROVED** [review]
@Rohit1173 LGTM.

**Comment by Rohit1173** [comment]
@rt4914 @anandwana001 PTAL

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  fun testAdministratorControlsFragment_clickEditProfile_checkSendingTheCorrectIntent() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        0
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.edit_profiles_text_view))
+        .perform(click())
+      Intents.intended(IntentMatchers.hasComponent(ProfileListActivity::class.java.name))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Use direct imports
`intended(hasComponent(ProfileListActivity::class.java.name))`

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+  @Test
+  fun testAdministratorControlsFragment_clickEditProfile_checkSendingTheCorrectIntent() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        0
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.edit_profiles_text_view))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Single line
`onView(withId(R.id.edit_profiles_text_view)).perform(click())`

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+  @Test
+  fun testAdministratorControlsFragment_clickAppVersion_checkSendingTheCorrectIntent() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        0
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.administrator_controls_list)).perform(
+        RecyclerViewActions.scrollToPosition<RecyclerView.ViewHolder>(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Ditto Direct imports
`scrollToPosition<RecyclerView.ViewHolder>(`

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+          3
+        )
+      )
+      onView(withId(R.id.app_version_text_view)).perform(click())
+      Intents.intended(IntentMatchers.hasComponent(AppVersionActivity::class.java.name))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Ditto

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Rohit1173 PTAL thanks.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt (hunk 4 above)
Comment (Rohit1173) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: done

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt (hunk 5 above)
Comment (Rohit1173) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: done

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt (hunk 6 above)
Comment (Rohit1173) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: done

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt (hunk 7 above)
Comment (Rohit1173) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: done

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+import org.oppia.android.util.networking.NetworkConnectionDebugUtilModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:86]: Add a comment above this:

```
/** Tests for [AdministratorControlsFragment]. */
```

**Review by anandwana001 — APPROVED** [review]
LGTM, left a nit suggestion.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt (hunk 8 above)
Comment (Rohit1173) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:86]: done

**Comment by Rohit1173** [comment]
@rt4914 changes done, PTAL.Thanks

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**